source/agora/test/Base.d

```diff
@@ -1656,6 +1665,12 @@ private mixin template TestNodeMixin ()
         return super.makeEnrollmentManager();
     }
 
+    /// Returns an instance of a DNSResolver
+    protected override DNSResolver makeDNSResolver ()
+    {
+        return new LocalRestDNSResolver(this.dns_chan);
```

Comment:
Why are we passing the `Channel` around instead of doing a registry lookup (Localrest registry, obv) ?

Reply:
Because `registry` only works for `Listener`s that require a `BindChn` which are only available for `spawn`d LocalRest threads (nodes). And LocalRest requires an interface

Reply:
Also, `FullNode`s dont have access to registry